Make archive sub attrib optional in packages sources attributes in package YAML file

alanoe opened the pull request.

Commit:
Infer git repo name from git repo URL in get_git_repository()

Before, it was either assumed that git repo name was identical
to other attribute with a different meaning (eg package name or
package tar.gz name) or it was extracted from git repo URL outside
get_git_repository().

Commit:
Make 'archive' subattribute optional in package sources attributes (git, hg, url) in package YAML file

snowcrshd commented on lib/repository.py:
```diff
@@ -41,6 +42,13 @@ def get_git_repository(name, remote_repo_url, parent_dir_path):
     """
+    # infer name if it is not provided
+    if not name:
+        match_obj = re.match(".*/([a-zA-Z0-9\-]*)\.git", remote_repo_url)
```
Won't this break if the repository name is some weird string?

There are some [special characters that are allowed in a URL](http://stackoverflow.com/questions/1856785/characters-allowed-in-a-url#1856809) that could show up in the repository's name.

olavphibm replied:
I'd use `urlparse` combined with `os.path.basename` for that.

alanoe replied:
imho [a-zA-Z0-9\-_]* is a safe regex, I have never seen other chars in source code repositories names

olavphibm commented:
start build

ghost commented:
I'm with either removing the "name" parameter completely, or leaving it as a keyword arg, so we don't have to pass `None` every time.

olavphibm commented:
start build

alanoe commented:
OH> I'm with either removing the "name" parameter completely, or leaving it as a keyword arg, so we don't have to pass None every time.

Agreed. I see no reason to have this argument in our build code, so I'll remove it.

olavphibm commented on lib/repository.py:
```diff
@@ -35,12 +36,15 @@ def __init__(self, push_info):
     """
+    # infer name
+    url_parts = urlparse.urlparse(remote_repo_url)
+    name = os.path.basename(url_parts.path)
```
This name might have a ".git" that we need to remove.

alanoe replied:
Done

snowcrshd replied:
I think that `# infer name` could be removed.

It doesn't add much information to that context. It ends up being just noise in the source code.

olavphibm commented on lib/repository.py:
```diff
@@ -35,12 +36,15 @@ def __init__(self, push_info):
     """
-    Get a local git repository located in a subdirectory of the parent
-    dir, according to specified name.
+    Get a local git repository located in a subdirectory of the parent directory.
```
You could mention it is "named after the base name of the URL path"

alanoe replied:
Done

olavphibm commented on lib/versions_repository.py:
```diff
@@ -15,12 +15,12 @@ def setup_versions_repository(config):
     anything fails.
     """
-    path, dir_name = os.path.split(
+    path, _ = os.path.split(
```
`os.path.dirname` could be used here instead of `split`

alanoe replied:
I prefer to not alter how parent directory path is obtained in this patch. This is an improvement which could be done in a separate patch.

fabianorosas commented:
start build

alanoe commented:
start build

snowcrshd commented on lib/repository.py:
```diff
@@ -35,12 +36,16 @@ def __init__(self, push_info):
+    named after the file name of the URL path (git default).
     If it does not exist, clone it from the specified URL.
     """
+    # infer name
```
Please remove this comment, it doesn't add meaningful information.

alanoe replied:
IMHO considering a comment to be obvious (and thus useless) or not is usually a matter of personal taste. Very minor issue IMHO. Anyway, I have updated the comment to make it more useful.

ghost commented:
start build

Commit:
Improve CLI arguments help

- Mention explicitly that repository is a git repository. This avoids
  confusion with packages repository (eg yum repository)
- Make it clearer that builds_version is not necessary a version number,
  but any git repository branch name

Commit:
Rename 'versions git repository' to 'packages metadata git repository' in doc

It is better to describe what the git repository contains than its name.
In the future, I suggest we also rename this git repository to packages_metadata
to make its purpose clearer.

ghost commented:
start build

farosas commented:
This relates to #38

alanoe commented:
Pls let us merge this ASAP so I do not need to rebase this patch again.
